### ISSUE TITLE
pass variable as initial symbol instead of underscore

### ISSUE THREAD

**Issue by akdienes** [jkrumbiegel/Chain.jl#49]
Hello,

I apologize if this has been addressed before. I like the syntax of Chain.jl but I was thinking it would be nice to be able to use the original variable name rather than only underscores. The front-page example
```
@chain df begin
  dropmissing
  filter(:id => >(6), _)
  groupby(:group)
  combine(:age => sum)
end
```

Could be something like

```
@chain df begin
  dropmissing
  filter(:id => >(6), df)
  groupby(:group)
  combine(:age => sum)
end
```

Let me know what you think.

**Comment by jkrumbiegel** [comment]
The df symbol stays valid throughout the expression so I can think it would be a bit confusing to update it with the line by line results. Also, in your case the result stays a dataframe but that isn't necessarily so, in that case it would be even more confusing, don't you think?

**Comment by akdienes** [comment]
good point. I think maybe the impetus for that idea was simply that I find the underscore to be very visually low-profile and sometimes hard to locate in an expression. Maybe what could be a better idea than my original is to allow custom-defined symbols as an alias for `_` ? Something like

```
@chain df begin as _tmp_
  dropmissing
  filter(:id => >(6), _tmp_)
  groupby(:group)
  combine(:age => sum)
end
```

Where here `as` would be a special keyword creating the alias. Obviously the exact semantics are not really important for the concept.
